api/v1alpha1/externaldns_types.go

```diff
@@ -113,6 +113,7 @@ type ExternalDNSDomain struct {
 	//
 	//  "Exclude": Exclude the domain set specified
 	//  by name or pattern.
+	// Only one "Pattern" filter types may be present for inclusion and exclusion.
```

Comment:
To elaborate on what I mean by combining patterns, take the "multiple regex include filter" test case for example; the input is this:
```go
			domainInput: []v1alpha1.ExternalDNSDomain{
				{
					ExternalDNSDomainUnion: v1alpha1.ExternalDNSDomainUnion{
						MatchType: v1alpha1.DomainMatchTypeRegex,
						Pattern:   pointer.StringPtr("(.*)\\.abc\\.com"),
					},
					FilterType: v1alpha1.FilterTypeInclude,
				},
				{
					ExternalDNSDomainUnion: v1alpha1.ExternalDNSDomainUnion{
						MatchType: v1alpha1.DomainMatchTypeRegex,
						Pattern:   pointer.StringPtr("(.*)\\.def\\.com"),
					},
					FilterType: v1alpha1.FilterTypeInclude,
				},
			},
```
Right now, this is expected to fail:
```go
			expectErr:     true,
			expectedError: "more than one domain pattern cannot be specified for inclusion",
```
We could just as easily take that input and produce the following output:
```go
			expectedArgs: []string{"--regex-domain-filter=(((.*)\\.abc\\.com)|((.*)\\.def\\.com))"},
```
That is, slap some parentheses around each pattern and join the parenthesized patterns using `|`.

BTW, you might want to use ``` `` ``` for regexps to conserve backslashes (e.g., ```Pattern:   pointer.StringPtr("(.*)\\.def\\.com"),``` → ```Pattern:   pointer.StringPtr(`(.*)\.def\.com`),```).

Reply:
Done, I've also added validation of the regex's so that they can be joined with some degree of certainty that the resulting expression is also valid. Also switched to backticks.

Reply:
This comment: "Only one 'Pattern' filter types may be present for inclusion and exclusion." can be deleted now.

Reply:
Fixed